### PR TITLE
[SandboxIR] Added setVolatile function to LoadInst and StoreInst

### PR DESCRIPTION
**Pull request by wizardengineer**
Prev PR had Branch conflicts: https://github.com/llvm/llvm-project/pull/101284, I decided to make a new one to circumvent any future issues. 

cc: @vporpo

### CODE REVIEW

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 1]
```diff
@@ -778,13 +784,23 @@ define void @foo(ptr %arg0, ptr %arg1) {
...
+  EXPECT_FALSE(NewVLd->isVolatile());
+  NewVLd->setVolatile(OrigVolatileValue);
   EXPECT_TRUE(NewVLd->isVolatile());
   EXPECT_EQ(NewVLd->getName(), "NewVLd");
   // Check create(InsertAtEnd)
   sandboxir::LoadInst *NewLdEnd =
       sandboxir::LoadInst::create(Ld->getType(), Arg1, Align(8),
                                   /*InsertAtEnd=*/BB, Ctx, "NewLdEnd");
   EXPECT_FALSE(NewLdEnd->isVolatile());
+  OrigVolatileValue = NewLdEnd->isVolatile();
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:799]: The only difference between `NewLdEnd` and `NewLd` is that it's at the end of the block. So I don't think we need to check volatile again.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 2]
```diff
@@ -797,6 +813,10 @@ define void @foo(ptr %arg0, ptr %arg1) {
                                   /*InsertAtEnd=*/BB,
                                   /*IsVolatile=*/true, Ctx, "NewVLdEnd");
   EXPECT_TRUE(NewVLdEnd->isVolatile());
+  OrigVolatileValue = NewVLdEnd->isVolatile();
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:816]: Same here, I would remove these checks.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 3]
```diff
@@ -858,6 +889,11 @@ define void @foo(i8 %val, ptr %ptr) {
       sandboxir::StoreInst::create(Val, Ptr, Align(8),
                                    /*InsertAtEnd=*/BB, Ctx);
   EXPECT_FALSE(NewStEnd->isVolatile());
+  OrigVolatileValue = NewStEnd->isVolatile();
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:892]: Same here, remove these checks.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 4]
```diff
@@ -870,6 +906,11 @@ define void @foo(i8 %val, ptr %ptr) {
                                    /*InsertAtEnd=*/BB,
                                    /*IsVolatile=*/true, Ctx);
   EXPECT_TRUE(NewVStEnd->isVolatile());
+  OrigVolatileValue = NewVStEnd->isVolatile();
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:909]: Same here, remove these checks.

File: llvm/unittests/SandboxIR/TrackerTest.cpp [hunk 5]
```diff
@@ -708,3 +708,35 @@ define void @foo(i8 %arg0, i8 %arg1, i8 %arg2) {
...
+  sandboxir::Context Ctx(C);
+
+  auto *F = Ctx.createFunction(&LLVMF);
+  auto *BB = &*F->begin();
+  auto It = BB->begin();
+  auto *Load = cast<sandboxir::LoadInst>(&*It++);
+  auto *Store = cast<sandboxir::StoreInst>(&*It++);
+
+  EXPECT_FALSE(Load->isVolatile());
+  EXPECT_FALSE(Store->isVolatile());
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/TrackerTest.cpp:730]: nit: I would move this line just before the Store checks at line 737.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 1 above)
Comment (medievalghoul) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp:799]: Good catch, i'll make sure to remove it.

File: llvm/unittests/SandboxIR/TrackerTest.cpp (hunk 5 above)
Comment (medievalghoul) [reply llvm/unittests/SandboxIR/TrackerTest.cpp:730]: fixed.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 4 above)
Comment (medievalghoul) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp]: done

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 1 above)
Comment (medievalghoul) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp]: done

File: llvm/unittests/SandboxIR/TrackerTest.cpp (hunk 5 above)
Comment (medievalghoul) [reply llvm/unittests/SandboxIR/TrackerTest.cpp:730]: the commit didn't go through. I have to commit again

Comment (medievalghoul) [reply llvm/unittests/SandboxIR/TrackerTest.cpp]: should be fine now

**Review by vporpo — APPROVED** [review]
Looks good, thanks for the patch! I will merge it once the bots pass.

**Pull request merged**